lib/fastly_nsq/http.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'net/https'
+require 'fastly_nsq/http/nsqd'
+require 'fastly_nsq/http/nsqlookupd'
+
+class FastlyNsq::Http
+  def initialize(uri:, cert_filename: ENV['NSQ_SSL_CERTIFICATE'], key_filename: ENV['NSQ_SSL_KEY'])
+    @uri = uri.is_a?(URI) ? uri : URI.parse(uri)
+    @cert_filename = cert_filename
+    @key_filename = key_filename
+  end
+
+  def get(data = nil)
+    if data
+      params = URI.encode_www_form(data)
+      uri.query = params
+    end
+    req = Net::HTTP::Get.new(uri.request_uri)
+    http.request(req)
+  end
+
+  def post(data, body = nil)
+    params = URI.encode_www_form(data)
+    uri.query = params
+    req = Net::HTTP::Post.new(uri.request_uri)
+    http.request(req, body)
+  end
+
+  def use_ssl
+    return unless can_use_ssl?
+    http.use_ssl = true
+    http.cert = cert
+    http.key = key
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
I thought it was a global sign cert but it might be self signed?

Reply:
For anything else you would have to make sure the local cert store is setup correct on each box this would be used on. The most annoying case would be for developers however This could easily be made to be configurable.

Reply:
+1 to configuration